### DISCUSSION TITLE
Question about disconnected waveguides

### DISCUSSION THREAD

**Discussion by pwflanigan** [SiEPIC/openEBL-2024-05#217] [Q&A] [answered]
I know I missed the deadline last week, but this has been on my mind, and getting the answer now will help me prepare for the next submission.

In my previous (pre-GitHub) submissions, I used Nazca design (https://nazca-design.org/) to lay out arrays of rectangles to make a sort of Bragg grating like in this figure:
![image](https://github.com/SiEPIC/openEBL-2024-05/assets/136763549/6eab1bbd-ff22-470a-838a-7fa74331cfa8)

When I tried to submit these Nazca-made GDS files, they didn't pass the design check because I got numerous

> Error: Shapes outside component: Shapes for device layers need to be inside a component. At minimum, they must be inside a cell and inside a DevRec shape. Read more about requirements for components: https://github.com/SiEPIC/SiEPIC-Tools/wiki/Component-and-PCell-Layout

messages (and disconnected pin messages, having to do with the vertical outcouplers and y-splitters that were manually inserted after the rectangular blocks were created). Note that the blocks were in cells that were nested under a single top cell.

I understand that the requirements have changed, but I'm wondering how you would suggest creating these kinds of disconnected waveguides, because the link in the error message does really go into that. I did find this: https://github.com/SiEPIC/SiEPIC_EBeam_PDK/blob/master/Documentation/Disconnected%20Waveguide/Summary_Disconnected_Waveguide.pdf, but I'm not sure if it will work for me because I need the waveguide to be disconnected on both sides. 

Thanks for the help.

**Comment by lukasc-ubc** [comment]
Bragg gratings are fine.  Here is an example cell from the EBeam PDK:
<img width="394" alt="image" src="https://github.com/SiEPIC/openEBL-2024-05/assets/15843200/42af0766-059b-424c-a49c-630e76485257">

You just need well structured cells with PinRec & DevRec layers as described in the link above.  

Disconnected waveguides is something completely different.  We don't want to have disconnected waveguides since they cause reflections, hence we have tests to verify this.

**Comment by pwflanigan** [comment]
OK, I understand. We were using the disconnected waveguides because our simulations showed that there were very effective polarizers, in that they blocked (reflected) TE light very well, but if the current PDK does not support that then we will consider something else. Let me think about it for a little bit, then I'll close the thread.

**Reply by pwflanigan** [reply]
To clarify, is it possible to parameterize this built-in Bragg grating component (i.e., have a user-controlled pitch, grating thickness, length, etc.)?

**Comment by lukasc-ubc** [comment, marked as answer]
Yes
